pdns/dnsdist.cc

```diff
@@ -2763,3 +3096,59 @@ int main(int argc, char** argv)
 #endif
   }
 }
```

Comment:
It took me a while to understand what exactly the router does, and how it interacts with the other components, so perhaps a few comments about the general XSK design/architecture would help?

Reply:
Ok, no problem. I added some comments.
The design here is indeed a bit complicated.
Since `boost::lockfree::spsc_queue` does not accept `unique_ptr<XskPacket>` as a parameter (`unique_ptr` cannot be copy constructed), I made `XskPacket` as the parameter of `boost::lockfree::spsc_queue`. But I am worried about the copying overhead of XskPacket, so XskFrameInfo and XskDelayPacketInfo are constructed instead of XskPacket in many cases to reduce the copying overhead.
The presence of `XskFrameInfo` and `XskDelayPacketInfo` adds a lot of complexity here.
I now use `XskPacket*` as an element in boost::lockfree::spsc_queue in my refactoring. In this way, the overhead of copying is eliminated, and `XskFrameInfo` and `XskDelayPacketInfo` are also eliminated. I also changed the name of `XskExtraInfo` to `XskWorker`, which may be more appropriate.
The code here is probably better understood now.
I also added some comments, but I'm not sure if my comments are enough. I can add more comments if needed.

Reply:
Thanks a lot, I'll have a look and do a new review tomorrow!